src/bidiMapper/domains/context/browsingContextProcessor.spec.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2021 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { StubTransport } from '../../../tests/stubTransport.spec';
+
+import * as chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import * as sinon from 'sinon';
+
+import { BrowsingContextProcessor } from './browsingContextProcessor';
+import { CdpConnection, CdpClient } from '../../../cdp';
+import { Context } from './context';
+import { BrowsingContext } from '../../bidiProtocolTypes';
+
+describe('BrowsingContextProcessor', function () {
+  let mockCdpServer: StubTransport;
+  let browsingContextProcessor: BrowsingContextProcessor;
+  let cdpConnection: CdpConnection;
+
+  const EVALUATOR_SCRIPT = 'EVALUATOR_SCRIPT';
+  const NEW_CONTEXT_ID = 'NEW_CONTEXT_ID';
+  const TARGET_ATTACHED_TO_TARGET_EVENT = {
+    method: 'Target.attachedToTarget',
+    params: {
+      sessionId: '_ANY_VALUE_',
+      targetInfo: {
+        targetId: NEW_CONTEXT_ID,
+        type: 'page',
+        title: '',
+        url: '',
+        attached: true,
+        canAccessOpener: false,
+        browserContextId: '_ANY_ANOTHER_VALUE_',
+      },
+      waitingForDebugger: false,
+    },
+  };
+
+  beforeEach(async function () {
+    mockCdpServer = new StubTransport();
+    cdpConnection = new CdpConnection(mockCdpServer);
+    browsingContextProcessor = new BrowsingContextProcessor(
+      cdpConnection,
+      'SELF_TARGET_ID',
+      sinon.fake as unknown as (t: Context) => Promise<void>,
```

Comment:
Can you wrap the `sinon.fake as unknown as (t: Context) => Promise<void>` code into a `typedFake` function similar to `typedSpy`?

Reply:
Also, shouldn't this call `fake` (e.g. `fake()`)?